packages/react-scripts/scripts/init.js

```diff
@@ -105,16 +105,44 @@ module.exports = function(
     '..'
   );
 
+  let templateJsonPath;
+  if (templateName) {
+    templateJsonPath = path.join(templatePath, 'template.json');
+  } else {
+    // TODO: Remove support for this in v4.
+    templateJsonPath = path.join(appPath, '.template.dependencies.json');
+  }
+
+  let templateJson = {};
+  if (fs.existsSync(templateJsonPath)) {
+    templateJson = require(templateJsonPath);
+  }
+
   // Copy over some of the devDependencies
   appPackage.dependencies = appPackage.dependencies || {};
 
   // Setup the script rules
-  appPackage.scripts = {
-    start: 'react-scripts start',
-    build: 'react-scripts build',
-    test: 'react-scripts test',
-    eject: 'react-scripts eject',
-  };
+  const templateScripts = templateJson.scripts || {};
+  appPackage.scripts = Object.assign(
+    {
+      start: 'react-scripts start',
+      build: 'react-scripts build',
+      test: 'react-scripts test',
+      eject: 'react-scripts eject',
+    },
+    templateScripts
+  );
+
+  // Update scripts for Yarn users
+  if (useYarn) {
+    appPackage.scripts = Object.entries(appPackage.scripts).reduce(
+      (acc, [key, value]) => ({
+        ...acc,
+        [key]: value.replace(/(npm run |npm )/, 'yarn '),
```

Comment:
Replace any `npm` commands for Yarn users. This is the same logic we use for replacing those commands in the README file.

Reply:
Do we do this for our scripts currently?

Reply:
Not sure that we need to be too smart about this.

Reply:
No, but I think this is a "nice to have" - as we're doing the same thing for the `README` in the same file.

It creates less confusion for people new to CRA too.

Reply:
I should clarify, we don't need this for _our_ scripts - this is to ensure that scripts created by others are normalised - as we do with README files. This allows people to create scripts with commands like the below:
```
"scripts": {
  "build-dependency": "some-script",
  "compile": "npm run build-dependency && npm run build",
}
```

We should also consider (in future) normalising across operating systems too. For example, you can't do `yarn build && yarn test` on Windows (without WSL). But that's too much at this stage.

Reply:
`npm run build-dependency && npm run build` will work on windows without WSL. Is that what you meant?

Reply:
OK, that's good to know - I searched briefly and got mixed results. Thanks @ianschmitz.